### DISCUSSION TITLE
Image | NanoPC T6

### DISCUSSION THREAD

**Discussion by quasistatic-setup** [MichaIng/DietPi#6400] [Image Request] [answered]
### Creating an image request
Currently only the images of the smaller brothers NanoPi R6S and R6C are available for download. Not for the big brother NanoPC T6.

#### Formal device information
- Device name | NanoPC T6
- Official product URL | https://www.friendlyelec.com/index.php?route=product/product&path=69&product_id=292
- Official Wiki URL | https://wiki.friendlyelec.com/wiki/index.php/NanoPC-T6
- Image download URL | https://www.friendlyelec.com/image/cache/catalog/details/T6_03-900x630.jpg

#### Is the SBC officially supported by the Debian installer?
- no

#### If not, is a reliable 3rd party Debian image available for this SBC?
- Debian Bullseye Core 5.10 ARM64
- Debian Bullseye Desktop 5.10 ARM64
- Debian Bullseye Minimal 5.10 ARM64
Images provided by manufacturer FriendlyElec
https://drive.google.com/drive/folders/1dQXz0RSyAFQCHYsjnkPt_jhL9PnGqwot

**Comment by MichaIng** [comment]
Many thanks for your request. FriendlyELEC provides the same images for R6S, R6C and T6, so our R6S/R6C image should work just fine. We may just change the name to "NanoPi R6S/R6C/T6" and remove the Ethernet LED udev rules on first boot, base on `/proc/device-tree/model`, as the T6 has no dedicated Ethernet LEDs like R6S and R6C have.

Do you have this SBC already so that you can test, or are you just thinking about ordering it?

I'll ask for a developer sample as well.

**Reply by quasistatic-setup** [reply]
Unfortunately, it doesn't work at all. I tried it with 3 different SD cards. I flashed the image with Balena. When I pop it in, I don't get any signal out of HDMI. It also does not show up in the network. My router does not see it at all. I guess it does not boot.

There are more physical differences between R6S/R6C and the T6. T6 has 3 HDMI ports. 2 out, 1 in. It has 2x 2.5 gbps network ports and no WAN port. SoC is also different. It has a Rockchip RK3588. The other ones have both Rockchip RK3588S.

Am I doing something wrong or could the difference in hardware cause the issue?

**Reply by MichaIng** [reply]
We use the FriendlyELEC images as basis, so basically it must work. But probably there was a needed bootloader update. Let me update our image to be based on the latest one from FriendlyELEC.

**Reply by MichaIng** [reply, marked as answer]
Can you please test the R6S image from here: https://dietpi.com/downloads/images/testing/

**Reply by quasistatic-setup** [reply]
Awesome, thanks for that. I tried _DietPi_NanoPiR6S-ARMv8-Bullseye.7z_ and it booted.
Now this device is usable for me. 
Bought the project a coffee via PayPal. :)
I guess this test image is not a stable version. When is there going to be the next stable version with the bootloader fix?

**Reply by MichaIng** [reply]
It has been built with stable DietPi code so can be seen as stable. The only thing I will change is removing the Ethernet LED udev rules on first boot, but that will have to wait for next DietPi release (as v8.18 will be released in the next hour).

Just to be sure: There are no dedicated Ethernet LEDs at the PCB, right, resp. there is no `wan_led` or `lan_led` device here, right?
```sh
ls -l /sys/class/leds/
```

> Bought the project a coffee via PayPal. :)

Many thanks, it arrived 🙂.

**Reply by quasistatic-setup** [reply]
Correct, it has no Wan or Lan indicator. Only sys and LED1. 

root@DietPi:~# ls -l /sys/class/leds/
total 0
lrwxrwxrwx 1 root root 0 Jun  3 11:17 ACT -> ../../devices/platform/leds/leds/ACT
lrwxrwxrwx 1 root root 0 Jun  3 11:17 PWR -> ../../devices/platform/leds/leds/PWR
lrwxrwxrwx 1 root root 0 Jun  3 11:17 default-on -> ../../devices/virtual/leds/default-on
lrwxrwxrwx 1 root root 0 Jun  3 11:17 mmc0 -> ../../devices/virtual/leds/mmc0
lrwxrwxrwx 1 root root 0 Jun  3 11:17 mmc0:: -> ../../devices/platform/emmc2bus/fe340000.mmc/leds/mmc0::

**Reply by MichaIng** [reply]
Two questions:
1. What is the output of `cat /proc/device-tree/model`?
2. Any preference for the hardware name on website in banner etc? "NanoPi R6S/R6C/T6" is probably a little long, but at least everyone finds its SBC name explicitly inside. Alternatively "NanoPi 6 series"?

First step done, assuming that "T6" will be part of `/proc/device-tree/model` https://github.com/MichaIng/DietPi/commit/1c7b0fc

**Reply by quasistatic-setup** [reply]
1. FriendlyElec NanoPC-T6
2. It is not a NanoPi, it is a NanoPC. So "NanoPi R6S/R6C/T6" is a compromise in which you can find T6. I think "NanoPi 6 series" is too broad.

**Reply by ghost** [reply]
This attitude is why people are reluctant to post comments. When i need help, or am just praising the work of other people i dont need the reply from a pedant. goodbye.

**Reply by MichaIng** [reply]
Huh? Totally fine to give an accurate reply. I was explicitly asking for it 🙂. Indeed it is "NanoPC", but using that as well like "NanoPi R6S/R6C/NanoPC T6" would be way too long, hence the compromise that we also use in case of "NanoPi M3/T3/Fire3" and "NanoPi M4/T4/NEO4". But I do not like it that much. The problem in case of the 6 series is that there are additionally two R variants. The M5 will surely be added soon, and NEO5 probably as well, which would make the name longer and longer, and of course we'd need to change it again and again.

"NanoPi 6 series" may be a bit broad, but actually this is what we need, as FriendlyELEC releases a broad range of SBCs for/with every SoC.

**Comment by MichaIng** [comment]
Done: https://github.com/MichaIng/DietPi/commit/3a096eb
